cpp/benchmarks/transpose/transpose.cpp

```diff
@@ -40,6 +42,17 @@ static void BM_transpose(benchmark::State& state)
     cuda_event_timer raii(state, true);
     auto output = cudf::transpose(input);
   }
+
+  // Collect memory statistics.
+  auto const bytes_read =
+    input.num_columns() * input.num_rows() * sizeof(cudf::id_to_type<column_type_id>);
```

Comment:
Do we need to worry about this product overflowing `size_type`? I think the number of columns and rows are both `size_type`, so their product could overflow before multiplying by the `sizeof`. Maybe need to cast everything to `std::size_t`?

Reply:
Good spotting. I think you are correct.

Reply:
I've verified that both `.num_columns()` and `.num_rows()` return a `cudf::size_type`. To avoid any potential ambiguities, I would cast it to `uint64_t`. The attached performance log in this PR remains Ok, given that the current benchmark uses `32768` for both row and column sizes. This value, when squared, fits within int32_t.